server/filestore.go

```diff
@@ -9504,8 +9505,9 @@ var dios chan struct{}
 // Used to setup our simplistic counting semaphore using buffered channels.
 // golang.org's semaphore seemed a bit heavy.
 func init() {
-	// Limit ourselves to a max of 4 blocking IO calls.
-	const nIO = 4
+	// Limit ourselves to a sensible number of blocking I/O calls. Range between
+	// 4-16 concurrent disk I/Os based on the number of CPU cores available.
+	nIO := min(16, max(4, runtime.GOMAXPROCS(-1)))
```

Comment:
I think capping at 16 sounds good, only thing is that gomaxprocs check happens here before auto maxprocs runs, so in container setups for this to apply it is needed to setup GOMAXPROCS explicitly instead

Reply:
Excellent point @wallyqs

We need to make sure we have adjusted for running in a container. @neilalexander could you do followup PR?